### ISSUE TITLE
Adding support for Javascript and/or TypeScript

### ISSUE THREAD

**Issue by kristianmandrup** [atomist/antlr-ts#4]
Any plans or work on this already? What is the best way to get started? There are multiple parser generators out there... Cheers!

I'd like to write Rugs for Javascript/Node and TypeScript.

PS: I'd love even more to join the Atomist team. Working on developer automation is the story of my life... what I've always wanted to do.

**Comment by kristianmandrup** [comment]
I found the [antlr4ts](https://github.com/tunnelvisionlabs/antlr4ts) project. Would that be useful to get going?

**Comment by kristianmandrup** [comment]
Made a branch here with what I have at this point. To be continued ;)

https://github.com/kristianmandrup/antlr-ts/tree/javascript

**Comment by johnsonr** [comment]
What would be really great would be to use the TypeScript parser and put a `TreeNode` implementation over it. This should probably be in a separate project from `antlr-ts` or `treepath-ts`. It would be then possible to run path expressions over TypeScript.

See for example the implementation of `MicrogrammarBasedFileParser` in `automation-client-ts`. Depending on what information is returned from the TypeScript parser, it would be possible to do replacement as well, using positions.

So I think that ANTLR is _not_ the solution for TypeScript. But it may be for JavaScript.

Thanks for looking at this!

**Comment by kristianmandrup** [comment]
Yes, also since there is no ANTLR grammar for TypeScript yet, only for JS.
I found it a bit tricky to figure out where to "enter" in this eco system. I'd like to experiment with the micro grammar approach, which seems to be an easier entry point so I will follow your hint. 
I wish you had a blog post on writing micro grammars, but I think the Readme, tests and such are enough to start hacking... Thanks.

I also think JS will in time be a "legacy language" and be replaced with TypeScript for development.

I just noticed  Graal VM, which allows for full language interop at almost zero cost, we live in amazing times!!!
